### PR TITLE
[Flava] Add ckpt loading and accuracy metric to finetuning

### PR DESCRIPTION
**Pull request by ankitade**
- Accuracy metric for finetuning
- Add checkpoint saving and best ckpt loading based on val accuracy
- Load pretrained ckpt by default in classification model
- make num gpus 1 in qnli.yaml

Test plan
python -m flava.finetune config=flava/configs/finetuning/qnli.yaml
(val acc : 0.8651)

Loaded model weights from checkpoint at /data/home/deankita/torchmultimodal/examples/flava-epoch=03-step=10000.ckpt
/data/home/deankita/miniconda/envs/flava/lib/python3.8/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:330: PossibleUserWarning: Using `DistributedSampler` with the dataloaders. During `trainer.validate()`, it is recommended to use `Trainer(devices=1)` to ensure each sample/batch gets evaluated exactly once. Otherwise, multi-device settings use `DistributedSampler` that replicates some samples to make sure all devices have same batch size in case of uneven inputs.
  rank_zero_warn(
Validation DataLoader 0: 100%|████████████████████████████████████████████████████████████████████████████████████████████████| 171/171 [00:54<00:00,  3.15it/s]
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃          Validate metric           ┃            DataLoader 0            ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ validation/accuracy/classification │         0.8651315569877625         │
│  validation/losses/classification  │         0.4168359339237213         │

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #132
* #131
* #106
* #105
* __->__ #119

Differential Revision: [D37444938](https://our.internmc.facebook.com/intern/diff/D37444938)

### CODE REVIEW

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37444938).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37444938).

**Comment by ankitade** [comment]
@ankitade has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37444938).

File: torchmultimodal/models/flava/flava_model.py [hunk 1]
```diff
@@ -209,6 +209,7 @@ def flava_model_for_classification(
     classifier_activation: Callable[..., nn.Module] = nn.ReLU,
     classifier_normalization: Optional[Callable[..., nn.Module]] = None,
     loss_fn: Optional[Callable[..., Tensor]] = None,
+    pretrained_model_key: Optional[str] = "flava_full",
```
Comment (ebsmothers) [inline torchmultimodal/models/flava/flava_model.py:212]: Just wondering, why do we want to add this as a param to `flava_model_for_classification`? Feels to me like this class should not care about the pretrained weights. I like TorchVision's [approach](https://github.com/pytorch/vision/blob/main/torchvision/models/resnet.py#L730-L757) for handling this, maybe we can do something similar?

Comment (ankitade) [reply torchmultimodal/models/flava/flava_model.py:212]: I am just doing this for "short term uniformity" with the flava_for_pretraining. But yeah i agree, tv's approach is nicer and something we can potentially follow

**Comment by ankitade** [comment]
@apsdehal I am going to merge this today. please feel free to leave comments either ways and ping me. I can address them in a separate PR if I end up merging before u review
